gantry/routes/prediction/prediction.py

```diff
@@ -0,0 +1,214 @@
+import asyncio
+import logging
+
+import aiosqlite
+
+from gantry.routes.prediction.current_mapping import pkg_mappings
+from gantry.util import k8s, spec
+
+logger = logging.getLogger(__name__)
+
+IDEAL_SAMPLE = 5
+DEFAULT_CPU_REQUEST = 1.0
+DEFAULT_MEM_REQUEST = 2 * 1_000_000_000  # 2GB in bytes
+EXPENSIVE_VARIANTS = {
+    "sycl",
+    "mpi",
+    "rocm",
+    "cuda",
+    "python",
+    "fortran",
+    "openmp",
+    "hdf5",
+}
```

Comment:
The way I got these is processing all builds and selecting the variants that have some association with increased memory and cpu usage. We could integrate that logic into here but I think it would be quite expensive, considering that this list is unlikely to change drastically. 

I would prefer either manual updating or with some sort of cron job but happy to consider all options

Reply:
I think an asynchronous background job is a great idea. I'm okay not requiring that in this PR on that however. @cmelone can you create an issue to remind us to write a background job to dynamically build these values before we forget?

Reply:
yup will do!